arch/src/aarch64/mod.rs

```diff
@@ -144,21 +146,27 @@ pub fn configure_system<T: DeviceInfoForFdt + Clone + Debug, S: ::std::hash::Bui
     virtio_iommu_bdf: Option<u32>,
     gic_device: &dyn GicDevice,
     numa_nodes: &NumaNodes,
+    dtb_path: Option<File>,
 ) -> super::Result<()> {
-    let fdt_final = fdt::create_fdt(
-        guest_mem,
-        cmdline,
-        vcpu_mpidr,
-        vcpu_topology,
-        device_info,
-        gic_device,
-        initrd,
-        pci_space_address,
-        numa_nodes,
-        virtio_iommu_bdf,
-    )
-    .map_err(|_| Error::SetupFdt)?;
 
+    let mut fdt_final: Vec<u8>;
```

Comment:
This does not need to be `mut`.

Also I suspect clippy will complain `needless-late-init` for this code.

This can be written instead as:

```
let fdt_final = if let Some(dtb_file) = dtb_path {
   fdt::fdt_file_to_vec(...)
} else {
    ...
}
```